From 56366ba56892d9de832ca86c95acb6674820d360 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Thu, 14 May 2015 19:14:53 -0400
Subject: [PATCH] enable secure_delete by default

This sanitizes deleted data by overwriting it with zeroes. It prevents a
non-root user from obtaining the deleted data even though it can still
linger on the underlying storage device.
---
 dist/Android.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dist/Android.mk b/dist/Android.mk
index 20b3a0a..7eabbdc 100644
--- a/dist/Android.mk
+++ b/dist/Android.mk
@@ -27,7 +27,8 @@ common_sqlite_flags := \
 	-DSQLITE_OMIT_BUILTIN_TEST \
 	-DSQLITE_OMIT_COMPILEOPTION_DIAGS \
 	-DSQLITE_OMIT_LOAD_EXTENSION \
-	-DSQLITE_DEFAULT_FILE_PERMISSIONS=0600
+	-DSQLITE_DEFAULT_FILE_PERMISSIONS=0600 \
+	-DSQLITE_SECURE_DELETE
 
 device_sqlite_flags := $(common_sqlite_flags) \
     -DSQLITE_ENABLE_ICU \
